File: 2021/day2/Puzzle0.java
package day2;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class Puzzle0 {
	public static void main(String[] argv) {
		try {
			var lines = Files.readAllLines( Path.of("./2021/day2/input") );
			String gammaRate = "", epsilonRate = "";

			for ( var index = 0; index < lines.get(0).length(); index++ ) {
				int zeros = 0, ones = 0;
				for (var line : lines) {
					switch (line.charAt(index)) {
						case '0' -> zeros++;
						case '1' -> ones++;
					}
				}
				if (zeros > ones)
					gammaRate = gammaRate + "0";
				else
					gammaRate = gammaRate + "1";
			}

			for ( var index = 0; index < lines.get(0).length(); index++ ) {
				int zeros = 0, ones = 0;
				for (var line : lines) {
					switch (line.charAt(index)) {
						case '0' -> zeros++;
						case '1' -> ones++;
					}
				}
				if ( zeros < ones )
					epsilonRate = epsilonRate + "0";
				else
					epsilonRate = epsilonRate + "1";
			}


			System.out.println( Integer.parseInt( gammaRate, 2 ) * Integer.parseInt( epsilonRate, 2 ) );

		} catch (IOException e) { e.printStackTrace(); }
	}
}

File: 2022/day0/Puzzle0.kt
package day0

import java.nio.file.Files
import java.nio.file.Path

fun main( argv: Array<String> ) {
    val elfs = mutableListOf(0)
    val lines = Files.readAllLines( Path.of("./2022/day0/input") )
    var index = 0

    lines.forEach { line ->
        if ( line == "" ) {
            elfs.add(0)
            index += 1
        } else
            elfs[index] += Integer.valueOf( line )
    }
    println( elfs.max() ) // 72070
}
